src/obj.rs

```diff
@@ -1151,7 +1139,375 @@ f 5/5 1/13 4/14 8/6
+      material_library: Some("cube.mtl".to_owned()),
+      objects: vec![
+        Object {
+          name: String::new(),
+          vertices:
+            vec!(
+              ( 1.0, -1.0, -1.0),
+              ( 1.0, -1.0,  1.0),
+              (-1.0, -1.0,  1.0),
+              (-1.0, -1.0, -1.0),
+              ( 1.0,  1.0, -1.0),
+              ( 1.0,  1.0,  1.0),
+              (-1.0,  1.0,  1.0),
+              (-1.0,  1.0, -1.0))
+            .into_iter()
+            .map(|(x, y, z)|
+              Vertex {
+                x: x as f64,
+                y: y as f64,
+                z: z as f64 })
+            .collect(),
+          tex_vertices:
+            vec!(
+              (1.004952, 0.498633),
+              (0.754996, 0.498236),
+              (0.755393, 0.248279),
+              (1.005349, 0.248677),
+              (0.255083, 0.497442),
+              (0.25548, 0.247485),
+              (0.505437, 0.247882),
+              (0.505039, 0.497839),
+              (0.754598, 0.748193),
+              (0.504642, 0.747795),
+              (0.505834, -0.002074),
+              (0.75579, -0.001677),
+              (0.005127, 0.497044),
+              (0.005524, 0.247088))
+            .into_iter()
+            .map(|(u, v)| TVertex { u: u, v: v, w: 0. })
+            .collect(),
+          normals : vec!(),
+          geometry: vec!(
+            Geometry {
+              material_name: Some("Material".to_owned()),
+              smooth_shading_group: 0,
+              shapes:
+                vec!(
+                  (3, 3, 0, 0, 1, 1),
+                  (3, 3, 1, 1, 2, 2),
+                  (5, 7, 4, 4, 7, 5),
+                  (5, 7, 7, 5, 6, 6),
+                  (1, 1, 0, 8, 4, 9),
+                  (1, 1, 4, 9, 5, 7),
+                  (2, 2, 1, 1, 5, 7),
+                  (2, 2, 5, 7, 6, 6),
+                  (3, 11, 2, 2, 6, 6),
+                  (3, 11, 6, 6, 7, 10),
+                  (7, 5, 4, 4, 0, 12),
+                  (7, 5, 0, 12, 3, 13))
+                .into_iter()
+                .map(|(vx, tx, vy, ty, vz, tz)|
+                  Shape {
+                    primitive:
+                      Triangle(
+                        (vx, Some(tx), None),
+                        (vy, Some(ty), None),
+                        (vz, Some(tz), None)),
+                    groups: vec!(),
+                  })
+                .collect(),
+            }
+          )
+        }
+      ]
+    });
+
+  assert_eq!(parse(test_case.to_owned()), expected);
+}
+
+#[test]
+fn test_cube_tex_vert_missing_vw() {
+  use self::Primitive::{ Triangle };
+
+  let test_case =
+r#"
+# Blender v2.71 (sub 0) OBJ File: 'cube.blend'
+# www.blender.org
+mtllib cube.mtl
+o Cube
+v 1.000000 -1.000000 -1.000000
+v 1.000000 -1.000000 1.000000
+v -1.000000 -1.000000 1.000000
+v -1.000000 -1.000000 -1.000000
+v 1.000000 1.000000 -0.999999
+v 0.999999 1.000000 1.000001
+v -1.000000 1.000000 1.000000
+v -1.000000 1.000000 -1.000000
+vt 1.004952
```

Comment:
The idea of a textured line confuses me, but if it's possible to make then I guess we should support it!